### PR TITLE
CVE-2010-1502 and CVE-2015-1269 and CVE-2015-6757

### PR DESCRIPTION
**Pull request by ALwolfman9**
CVE's all researched and updated

### CODE REVIEW

File: cves/CVE-2010-1502.yml [hunk 1]
```diff
@@ -15,9 +15,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (samshiffman321) [inline cves/CVE-2010-1502.yml]: the description needs to have more detail aim for 1-2 full paragraphs explaining the vulnerability

File: cves/CVE-2010-1502.yml [hunk 2]
```diff
@@ -105,7 +110,7 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: I did not see any major events during this time.
```
Comment (samshiffman321) [inline cves/CVE-2010-1502.yml]: not sure this is necessary. if there were no major events I believe you can leave this blank

File: cves/CVE-2010-1502.yml [hunk 3]
```diff
@@ -172,3 +177,14 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer: 
+    The major coding mistake was that for two years, chromium did not sanitize its url
+    input, allowing for attackers to access local files.
+
+    The vulnerability was introduced in the initial commit, and was ignored up until it
```
Comment (samshiffman321) [inline cves/CVE-2010-1502.yml]: how do you know it was ignored? are you sure? is it possible that they just didn't know about it until then?

File: cves/CVE-2015-1269.yml [hunk 4]
```diff
@@ -15,7 +15,8 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Attackers can bypass HSTS restrictions due to a lack of canonicalizing
```
Comment (samshiffman321) [inline cves/CVE-2015-1269.yml]: this description needs to have more detail about the vulnerability in it. aim for 1-2 full paragraphs

File: cves/CVE-2015-6757.yml [hunk 5]
```diff
@@ -15,7 +15,8 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Trying to access data in ServiceWorker after it has been freed, creating a
```
Comment (samshiffman321) [inline cves/CVE-2015-6757.yml]: again, the descriptions of these vulnerabilities needs to be more detailed. aim for 1-2 full paragraphs

File: cves/CVE-2015-6757.yml [hunk 6]
```diff
@@ -160,6 +160,9 @@ lessons:
   complex_inputs:
     applies: 
     note: 
+  Basic_Coding_Practices:
+    applies: true
+    note: Making sure basic coding practices are followed, like not trying to acess data after it has been freed, will lead to better security.
```
Comment (samshiffman321) [inline cves/CVE-2015-6757.yml]: spelling error: should be "access" not "acess"

File: cves/CVE-2015-1269.yml [hunk 7]
```diff
@@ -174,3 +173,8 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer: 
+    The mistake that was made was that the developers tried to rely on whiteboxing
+    alone and didn't attempt to normalize any inputs.
```
Comment (gregmullin) [inline cves/CVE-2015-1269.yml]: This seems to be the cause of the vulnerability, I think you could extrapolate more from this though. Why do you think they didn't attempt to normalize input. Do you think they planned to and made a mistake, or did decide that it was unnecessary during the design process?

File: cves/CVE-2015-1269.yml [hunk 8]
```diff
@@ -34,7 +35,8 @@ fixes_vcc_instructions: |
 fixes:
 - :commit: c2dd599bdd35753da237172dc37a447e9d65c463
   :note: ''
-vccs: []
+vccs: 
+  - 0653ece4f182ab947d6809c311dfd4e8febaee1f
 upvotes_instructions: |
```
Comment (gregmullin) [inline cves/CVE-2015-1269.yml]: +5 upvotes from me. not sure how to add them in.

File: cves/CVE-2010-1502.yml [hunk 9]
```diff
@@ -54,8 +59,8 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer:
```
Comment (huntercaskey) [inline cves/CVE-2010-1502.yml]: Try to summarize the fix that they made to the vulnerability in a sentence or two.

File: cves/CVE-2010-1502.yml (hunk 1 above)
Comment (huntercaskey) [inline cves/CVE-2010-1502.yml]: Agreed. Try to explain a little bit more detail about the vulnerability while staying at a high-level. The readers should have a good understanding about what vulnerability entails from reading this.

File: cves/CVE-2010-1502.yml [hunk 10]
```diff
@@ -93,10 +97,11 @@ interesting_commits:
...
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 3d0baa144bed599d8a1f68afb45273de4901fde9
+    note: Changed chrome_url_data_manager into a singleton
+  - commit: 576a4ca9a1bfbcb6e255e9842db1ac3da62f8496
+    note: Six days later reverted back from singleton change, calling it
+          unnessessary.
```
Comment (huntercaskey) [inline cves/CVE-2010-1502.yml]: What was their rationale for considering the application of the Singleton pattern unnecessary?

File: cves/CVE-2015-1269.yml [hunk 11]
```diff
@@ -109,10 +109,8 @@ major_events:
     we want to capture what the development team was dealing with at the time.
   answer: 
   events:
-  - name: 
-    date: 
-  - name: 
-    date: 
+  - name: Treat HSTS and HPKP state independently.
```
Comment (huntercaskey) [inline cves/CVE-2015-1269.yml]: Make the researchers job a bit easier by providing the full names for terms followed by the acronyms.

File: cves/CVE-2015-1269.yml [hunk 12]
```diff
@@ -132,8 +130,9 @@ lessons:
     applies: 
     note: 
   least_privilege:
-    applies: 
-    note: 
+    applies: true
```
Comment (huntercaskey) [inline cves/CVE-2015-1269.yml]: I agree that least privilege applies here. Consider whether path traversal should also be included in this section

File: cves/CVE-2015-1269.yml (hunk 8 above)
Comment (huntercaskey) [inline cves/CVE-2015-1269.yml]: +3 Upvotes

**Pull request merged**